src/erlfmt_scan.erl

```diff
@@ -166,10 +168,14 @@ eof(undefined, Loc) ->
     {{eof, Loc}, undefined}.
 
 -spec last_node_string(state()) -> {unicode:chardata(), anno()}.
+last_node_string(#state{original = [{'#', _}, {'!', _} | _] = Tokens}) ->
+    split_shebang(Tokens);
 last_node_string(#state{original = [First | _] = Tokens}) ->
     String = stringify_tokens(Tokens),
     Location = erl_scan:location(First),
-    {String, token_anno([{text, String}, {location, Location}])}.
+    {String, token_anno([{text, String}, {location, Location}])};
+last_node_string(#state{original = []}) ->
+    {"", undefined}.
```

Comment:
When can this option happen?

Reply:
It is needed for these cases
```erlang

erlfmt:contains_pragma_string(
    "file.escript",
    "#! /usr/bin/env escript\n"
    "\n"
    "%% @format\n"
    "\n"
    "main(_) -> ok.\n"
)
erlfmt:contains_pragma_string(
    "file.escript",
    "#! /usr/bin/env escript\n"
    "\n"
    "main(_) -> ok.\n"
)
```

Reply:
Good catch!

I think this is because we're not setting `original` correctly when reading the shebang in [here](https://github.com/WhatsApp/erlfmt/blob/b95bc12e037001da4bc447f742c0f06c4dfebf8a/src/erlfmt_scan.erl#L92). I think a nicer fix would be to modify `split_shebang` to return the tokens, setting `original` in there and we wouldn't need any special-casing in here.

Reply:
Yes, this is amazing.  That for the great idea :D